web/app/services/auth.service.spec.ts

```diff
@@ -0,0 +1,84 @@
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
+import {inject, TestBed} from '@angular/core/testing';
+import {mockLoginResponse} from '../common/test_helpers/mock_login_data';
+import {AuthService, LoginRequest, LoginResponse} from './auth.service';
+
+const FAKE_LOGIN_REQUEST: LoginRequest = {
+  username: 'tacoRocat',
+  password: 'wholetthedogsout'
+};
+
+describe('AuthService', () => {
+  let mockHttp: HttpTestingController;
+  let authService: AuthService;
+  let mockStorage = {};
+
+  // TODO: generalize this as a test helper
+  const mockLocalStorage = {
+    getItem: (key: string): string => {
+      return key in mockStorage ? mockStorage[key] : null;
+    },
+    setItem: (key: string, value: string) => {
+      mockStorage[key] = `${value}`;
+    },
+    removeItem: (key: string) => {
+      delete mockStorage[key];
+    },
+    clear: () => {
+      mockStorage = {};
+    }
+  };
+
+  beforeEach(() => {
+    spyOn(localStorage, 'getItem').and.callFake(mockLocalStorage.getItem);
+    spyOn(localStorage, 'setItem').and.callFake(mockLocalStorage.setItem);
+
+    TestBed.configureTestingModule(
+        {imports: [HttpClientTestingModule], providers: [AuthService]});
+
+    mockHttp = TestBed.get(HttpTestingController);
+    authService = TestBed.get(AuthService);
+  });
+
+  afterEach(() => {
+    mockLocalStorage.clear();
+  });
+
+  it('should should make login request to backend', () => {
+    let loginResponse: LoginResponse;
+    authService.login(FAKE_LOGIN_REQUEST).subscribe((response) => {
+      loginResponse = response;
+    });
+
+    const loginRequest = mockHttp.expectOne('/data/login');
```

Comment:
it should be just  `/login` (?)

Reply:
So we should expose the whole json API under the `/api` prefix?

Reply:
I'm good with that. Let's start with `/login`. Leave the rest in `/data` for now

Reply:
We can change this in the front-end once the backend is ready